client/core/trade.go

```diff
@@ -2982,7 +3001,39 @@ func (t *trackedTrade) confirmRedemption(match *matchTracker) (bool, error) {
 			t.dc.log.Errorf("Failed to update match in db %v", err)
 		}
 		return false, errors.New("swap was already refunded by the counterparty")
-	} else if err != nil {
+
+	case errors.Is(err, asset.ErrTxRejected):
+		match.redemptionRejected = true
+		// If we're the maker, give up. They don't have the secret anyway.
+		if match.Side == order.Maker {
+			match.MetaData.Proof.SelfRevoked = true
+			return false, fmt.Errorf("%s transaction %s was rejected. We're the maker, so we're giving up",
+				unbip(toWallet.AssetID), coinIDString(toWallet.AssetID, redeemCoinID))
+		}
+		// We need to seek user approval before trying again, since new fees
+		// could be incurred.
+		actionRequest, note := newRejectedRedemptionNote(toWallet.AssetID, t.ID(), redeemCoinID)
+		t.notify(note)
+		c.requestedActionMtx.Lock()
+		c.requestedActions[dex.Bytes(redeemCoinID).String()] = actionRequest
+		c.requestedActionMtx.Unlock()
+		return false, fmt.Errorf("%s transaction %s was rejected. Seeking user approval before trying again",
+			unbip(toWallet.AssetID), coinIDString(toWallet.AssetID, redeemCoinID))
+	case errors.Is(err, asset.ErrTxLost):
+		// The transaction was nonce-replaced or otherwise lost without
+		// rejection or with user acknowlegement. Try again if we're the taker.
+		// If we're the maker we'll give up and try to refund.
+		c.log.Infof("Redemption %s (%s) has been noted as lost.", match.MetaData.Proof.TakerRedeem, unbip(toWallet.AssetID))
+		if match.Side == order.Taker {
+			match.MetaData.Proof.TakerRedeem = nil
+			match.Status = order.MakerRedeemed
+		}
```

Comment:
In this case the counterparty of the maker could have also got the secret. You didn't set `match.MetaData.Proof.SelfRevoked = true` over there though.. so will it refund?

Reply:
I've added commented tests for all actions now.